src/glue/jobs/s3_to_json.py

```diff
@@ -643,6 +646,41 @@ def _upload_file_to_json_dataset(
         os.remove(file_path)
     return s3_output_key
 
+def merge_dicts(x: dict, y: dict) -> Generator:
```

Comment:
Nit: Just out of curiosity, do you know if there is a feature that already exists from some package?

Reply:
Not in the standard library afaik. IMO, it's not worth adding another dependency.